tests/functional/base.py

```diff
@@ -17,8 +20,105 @@
 from custodia.server.config import parse_config
 
 
+def wait_pid(process, wait):
+    timeout = time.time() + wait
+    while time.time() < timeout:
+        pid, _ = os.waitpid(process.pid, os.WNOHANG)
+        if pid == process.pid:
+            return True
+        time.sleep(0.1)
+    return False
+
+
+def wait_socket(process, custodia_socket, wait):
+    timeout = time.time() + wait
+    while time.time() < timeout:
+        if process.poll() is not None:
+            raise AssertionError(
+                "Premature termination of Custodia server")
+        try:
+            s = socket.socket(family=socket.AF_UNIX)
+            s.connect(custodia_socket)
+        except OSError:
+            pass
+        else:
+            return True
+        time.sleep(0.1)
+    raise OSError('Timeout error')
+
+
+def translate_meta_uid(meta_uid):
+    current_uid = None
+
+    if meta_uid == "correct_id":
+        current_uid = pwd.getpwuid(os.geteuid()).pw_uid
+
+    if meta_uid == "incorrect_id":
+        actual_uid = pwd.getpwuid(os.geteuid()).pw_uid
+        for uid in [x.pw_uid for x in pwd.getpwall()]:
+            if uid != actual_uid:
+                current_uid = uid
+                break
+
+    if meta_uid == "correct_name":
+        current_uid = pwd.getpwuid(os.geteuid()).pw_name
+
+    if meta_uid == "incorrect_name":
+        actual_name = pwd.getpwuid(os.geteuid()).pw_name
+        for name in [x.pw_name for x in pwd.getpwall()]:
+            if name != actual_name:
+                current_uid = name
+                break
+
+    if meta_uid == "ignore":
+        current_uid = -1
+
+    return current_uid
+
+
+def translate_meta_gid(meta_gid):
+    current_gid = None
+
+    if meta_gid == "correct_id":
+        current_gid = grp.getgrgid(os.getegid()).gr_gid
+
+    if meta_gid == "incorrect_id":
+        actual_user = pwd.getpwuid(os.geteuid()).pw_name
+        actual_gid = grp.getgrgid(os.getegid()).gr_gid
+        for gid in [g.gr_gid for g in grp.getgrall() if
+                    actual_user not in g.gr_mem]:
+            if gid != actual_gid:
+                current_gid = gid
+                break
+
+    if meta_gid == "correct_name":
+        current_gid = grp.getgrgid(os.getegid()).gr_name
+
+    if meta_gid == "incorrect_name":
+        actual_user = pwd.getpwuid(os.geteuid()).pw_name
+        actual_group = grp.getgrgid(os.getegid()).gr_name
+        for name in [g.gr_name for g in grp.getgrall() if
+                     actual_user not in g.gr_mem]:
+            if name != actual_group:
+                current_gid = name
+                break
+
+    if meta_gid == "ignore":
+        current_gid = -1
+
+    return current_gid
+
+
+class UniqueNumber(object):
+    unique_number = 0
+
+    def get_unique_number(self):
```

Comment:
The function is not necessary. If you add ``request`` to any test argument list, then you can access the test name as ``request.node.name``. The name also contains the parameter values. This gives you an unique name that reflects the test case, too.